File: packages/swift/Sources/CatalystSwift/Models/CatalystAlbumBook.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

import Foundation

/// A generated photo/record book for an album or smart album.
///
/// `template` is one of "photo-book" | "record-book".
/// `quality` is one of "web" | "print".
/// `tocType` is one of "per-post" | "page-only" | nil.
/// `status` is one of "pending" | "processing" | "completed" | "failed".
public struct CatalystAlbumBook: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let albumId: String?
  public let smartAlbumId: String?
  public let template: String
  public let quality: String
  public let coverImageUrl: String?
  public let subtitle: String?
  public let customText: String?
  public let showBody: Bool
  public let showDate: Bool
  public let showUrl: Bool
  public let tocType: String?
  public let colophonText: String?
  public let status: String
  public let errorMessage: String?
  public let downloadUrl: String?
  public let expiresAt: Date?
  public let createdAt: Date?
  public let updatedAt: Date?
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystAlbumBook.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable
import kotlin.time.ExperimentalTime

/**
 * A generated photo/record book for an album or smart album
 *
 * [template] is one of "photo-book" | "record-book"
 * [quality] is one of "web" | "print"
 * [tocType] is one of "per-post" | "page-only" | null
 * [status] is one of "pending" | "processing" | "completed" | "failed"
 */
@Serializable
data class CatalystAlbumBook @OptIn(ExperimentalTime::class) constructor(
    val id: String,
    val albumId: String? = null,
    val smartAlbumId: String? = null,
    val template: String,
    val quality: String,
    val coverImageUrl: String? = null,
    val subtitle: String? = null,
    val customText: String? = null,
    val showBody: Boolean,
    val showDate: Boolean,
    val showUrl: Boolean,
    val tocType: String? = null,
    val colophonText: String? = null,
    val status: String,
    val errorMessage: String? = null,
    val downloadUrl: String? = null,
    val expiresAt: kotlin.time.Instant? = null,
    val createdAt: kotlin.time.Instant? = null,
    val updatedAt: kotlin.time.Instant? = null
)
